plugins/analysis/webinterface/httpserver/static/js/main.js

```diff
@@ -127,212 +126,121 @@ class Frontend {
         document.getElementById("out").innerHTML = "";
     }
 
-    displayNodesOnline(nodesOnline) {
+    displayNodes(nodes) {
         // this line might be a performance killer!
-        nodesOnline = Array.from(nodesOnline).sort();
+        nodes = Array.from(nodes).sort();
+        // When the active node gets removed, we should reset the nodeLink too
+        var isActiveNodePresent = false;
 
         let html = [];
-        for(let n of nodesOnline) {
+        for(let n of nodes) {
             if(n == this.activeNode) {
                 html.push('<span class="n active">' + n + "</span>");
+                isActiveNodePresent = true;
             } else {
                 html.push('<span class="n">' + n + "</span>");
             }
         }
         document.getElementById("nodesOnline").innerHTML = html.join("");
+        if (!isActiveNodePresent) {
+            // active node was removed
+            this.resetActiveNode();
+        }
     }
 }
 
 class Datastructure {
     constructor(app) {
         this.app = app;
         this.nodes = new Set();
-        this.nodesOnline = new Set();
-        this.nodesOffline = new Set();
-        this.nodesDisconnected = new Set();
         this.connections = new Set();
         this.neighbors = new Map();
+        this.recRemove = 0;
+        this.recDisconnect = 0;
```

Comment:
What exactly is this doing? Is it used? Maybe add a quick comment

Reply:
I forgot to delete them, they were here for debugging purposes only.